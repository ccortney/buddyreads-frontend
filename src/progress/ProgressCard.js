import { Button, Typography, Card, CardContent } from "@mui/material";
import { CircularProgressbar } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import RatingDialog from "../rating/RatingDialog";
import Rating from '@mui/material/Rating';
import { StarBorderOutlined, StarOutlined } from "@mui/icons-material";

/** Show ProgressCard component. 
 * 
 * Is rendered by Progress to show a "card" for each user.
 * Each card renders Rating component from Material UI and
 * RatingDialog when progress is 100%
 *
 * {Progress} -> {ProgressCard} -> { RatingDialog }
 */

const ProgressCard = ({user, progress, color, percent, setShowForm, isCurrentUser, rating, setRating, ratingOpen, setRatingOpen, handleRatingOpen, buddyRead, bookData}) => {
    
    return (
        <Card 
            sx={{
                bgcolor: color, 
                m: 2, 
                mt: 0,
                borderRadius: '0.7rem',
                color: 'white', 
                width: '126px' 
            }}>  
            <CardContent>
                <Typography>
                    {user.firstName}
                </Typography>
                <Typography variant="subtitle2" sx={{fontWeight: 'light'}}>
                    on Page {progress}
                </Typography>
            </CardContent> 

            <CardContent sx={{width: '5rem', pt: 0}}>
                <CircularProgressbar 
                value={percent} 
                text={`${percent}%`}
                styles={{
                    width: '4rem',
                    path: {stroke: "white", strokeWidth: '5px', filter: 'drop-shadow(1px 1px 1px white)'}, 
                    trail: {display: 'none'}, 
                    text: {fill: 'white'}
                }}
                />
            </CardContent>
            {isCurrentUser && percent!== 100 ? 
                <Button 
                    sx={{p: 1, color: 'white', size: 'small'}}
                    onClick = {() => setShowForm(true)}
                >
                    Update
                </Button>
                : null
            }
            {rating ?
                <Rating
                    name="read-only"
                    value={rating}
                    readOnly
                    icon={<StarOutlined fontSize="inherit" color="inherit"/>}
                    emptyIcon={<StarBorderOutlined fontSize="inherit" sx={{color: 'white'}}/>}
                />
                : null
            }
            {isCurrentUser && progress === bookData.pageCount ?
                <Button onClick={handleRatingOpen} sx={{color: "white", display: 'block'}}>
                    Update Rating
                </Button> 
            : null
            }
            {isCurrentUser ? 
                <RatingDialog 
                    bookData={bookData}
                    buddyRead={buddyRead}
                    ratingOpen={ratingOpen} 
                    setRatingOpen={setRatingOpen}
                    handleRatingOpen={handleRatingOpen} 
                    rating={rating}
                    setRating={setRating}
                />
            : null
            }
        </Card>
    )
}

export default ProgressCard;