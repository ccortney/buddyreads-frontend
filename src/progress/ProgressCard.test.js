import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import ProgressCard from "./ProgressCard"

const buddyRead = {
    id: 1, 
    createdBy: 1, 
    buddy: 2, 
    status: 'pending'
}

const user = {
    id: 10, 
    firstName: 'first', 
    lastName: 'last'
}

const bookData = {
    pageCount: 100
}

it("renders without crashing", function() {
    render(
        <MemoryRouter>
                <ProgressCard 
                    user={user}
                    progress={5}
                    color="secondary.main"
                    percent={100}
                    isCurrentUser={false}
                    rating={2}
                    ratingOpen={false}
                    bookData={bookData}
                    buddyRead={buddyRead}
                    />
        </MemoryRouter>
    );
  });

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
        <ProgressCard 
            user={user}
            progress={5}
            color="secondary.main"
            percent={100}
            isCurrentUser={false}
            rating={2}
            ratingOpen={false}
            bookData={bookData}
            buddyRead={buddyRead}
        />
    </MemoryRouter>
);;
  expect(asFragment()).toMatchSnapshot();
});