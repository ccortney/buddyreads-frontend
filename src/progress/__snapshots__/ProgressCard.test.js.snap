// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-oga8ba-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        first
      </p>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-q0efxh-MuiTypography-root"
      >
        on Page 5
      </h6>
    </div>
    <div
      class="MuiCardContent-root css-5emmk8-MuiCardContent-root"
    >
      <svg
        class="CircularProgressbar "
        data-test-id="CircularProgressbar"
        viewBox="0 0 100 100"
      >
        <path
          class="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="display: none; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
        />
        <path
          class="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="stroke: white; stroke-width: 5px; filter: drop-shadow(1px 1px 1px white); stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
        />
        <text
          class="CircularProgressbar-text"
          style="fill: white;"
          x="50"
          y="50"
        >
          100%
        </text>
      </svg>
    </div>
    <span
      aria-label="2 Stars"
      class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
      role="img"
    >
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ykz3gq-MuiSvgIcon-root"
            data-testid="StarBorderOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ykz3gq-MuiSvgIcon-root"
            data-testid="StarBorderOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ykz3gq-MuiSvgIcon-root"
            data-testid="StarBorderOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
